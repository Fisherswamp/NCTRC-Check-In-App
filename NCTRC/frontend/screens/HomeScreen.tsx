import React, {useState} from 'react';
import { SafeAreaView, Platform, StyleSheet, Text, View, Image, Button, TouchableOpacity, DatePickerIOSComponent, DatePickerIOSBase } from 'react-native'
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../index';
type HomeScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.Home>;
import { styles } from './Styles';

interface HomeScreenProps {
    navigation: HomeScreenNavigationProps;
}




const HomeScreen: React.FunctionComponent<HomeScreenProps> = (props) => {
    let { navigation } = props;
      return (
        <SafeAreaView style={styles.container}>
            <View style={styles.homeContainer}>
                <Image source={require('./../assets/NCTRClogo.png')} style={{ width: 300, height: 300 }}></Image>
                <TouchableOpacity style={styles.solidButton}onPress={() => navigation.navigate(AppScreens.Reason)}><Text style={styles.buttonText}>
                Check In
                 </Text>  
                </TouchableOpacity>
                <TouchableOpacity style={styles.solidButton}onPress={() => navigation.navigate(AppScreens.CheckOut)}><Text style={styles.buttonText}>
                Check Out
                 </Text>  
                </TouchableOpacity>
                <View style={styles.homeDiv}></View>
                <Button color='#884633'title="Sign in as Admin"onPress={() => navigation.navigate(AppScreens.AdminPass)} />

            </View>
       </SafeAreaView>
    );
};
export default HomeScreen;
