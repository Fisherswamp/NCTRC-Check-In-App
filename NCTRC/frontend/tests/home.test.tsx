
import React from "react";
import  HomeScreen from '../screens/HomeScreen';
import { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
    navigation: {
        addListener: jest.fn(), 
        canGoBack: jest.fn(), 
        dangerouslyGetParent: jest.fn(), 
        dangerouslyGetState: jest.fn(), 
        dispatch: jest.fn(), 
        goBack: jest.fn(), 
        isFocused: jest.fn(), 
        navigate: jest.fn(), 
        pop: jest.fn(), 
        popToTop: jest.fn(), 
        push: jest.fn(), 
        removeListener: jest.fn(), 
        replace: jest.fn(), 
        reset: jest.fn(), 
        setOptions: jest.fn(), 
        setParams: jest.fn(), 
    }, 
}

describe('Homescreen renders', () => {
    it('App render', () => {
      const tree = renderer.create(<HomeScreen navigation={props.navigation} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


describe('HomeScreen elements', () => {
    it("Home Screen elements", () => {
       let wrapper =  shallow(<HomeScreen navigation={props.navigation} />);
       expect(wrapper.find('Button').length).toBe(1); 
       expect(wrapper.find('Image').length).toBe(1); 
       expect(wrapper.find('Text').length).toBe(2); 
       expect(wrapper.find('View').length).toBe(2); 
    })
   
});

jest.useFakeTimers(); 