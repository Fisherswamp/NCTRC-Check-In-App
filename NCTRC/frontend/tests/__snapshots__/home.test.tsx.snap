// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homescreen renders App render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 300,
          "width": 300,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 20,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Check In
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 20,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Check Out
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 150,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#884633",
              },
            ]
          }
        >
          Sign in as Admin
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
